File: Server/updateBadges.php
<?php
require "globalFunctions.php";
$pdo = new PDO("pgsql:host=" . HOST . ";port=" . PORT . ";dbname=" . DATABASE . ";user=".USERNAME . ";password=" . PASSWORD);

$myId = $_POST["myId"];
$token = $_POST["token"];
$badges = $_POST["badges"];

$isValid =
    isAuthenticated($myId, $token) &&
    isValidInt($badges, 0, 30000);
if (!$isValid) {
    $pdo = null;
    die("Invalid");
}

$query = "update account set badges = ? where user_id = ?;";
$stmt = $pdo->prepare($query);
$stmt->execute(array($badges, $myId));

$pdo = null;

File: Server/updateBadges.go
package main

import (
	"context"
	"fmt"
	"net/http"
)

func updateBadges(w http.ResponseWriter, r *http.Request) {
	r.ParseForm()
	token := r.Form.Get("token")
	badges := r.Form.Get("badges")
	myId := getMyId(token)

	isValid :=
		isAuthenticated(myId) &&
			isValidInt(badges, 0, 30000)
	if !isValid {
		fmt.Fprintln(w, "Invalid")
		return
	}

	_, err := pool.Exec(context.Background(), "update account set badges = $1 where user_id = $2;", badges, myId)
	handleError(err)
}
